dvc/commands/dag.py

```diff
@@ -198,6 +464,30 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Print output files instead of stages.",
     )
+    dag_parser.add_argument(
+        "--direction",
+        choices=["LR", "TD"],
+        default="TD",
+        help=(
+            "Direction of the rendered mermaid DAG. "
+            "Can either be 'LR' for left-to-right or 'TD' for top-down'."
+        ),
+    )
+    dag_parser.add_argument(
+        "--status",
+        action="store_true",
+        default=False,
+        help=(
+            "Check the status of stages in the DAG. "
+            "(Only compatible with --mermaid and --md)"
+        ),
+    )
+    dag_parser.add_argument(
+        "--status-import",
+        action="store_true",
+        default=False,
+        help="Check the dependencies of import stages. (Only compatible with --status)",
+    )
```

Comment:
I'm not sure the naming of the option is perfect, and maybe the default behavior should be inverted. But I do think it is quite relevant to have it in one way or another.

In our case, we are importing large datasets from a data repo into a model repo. If we didn't skip the dependency check for these import stages, instead of ~2 seconds, we would need ~10 minutes to generate the DAG with stage statuses.

I think, actually, checking the status of imported deps should be sped up in the first place. But for now this options gives us at least a way around it.

Reply:
> But I do think it is quite relevant to have it in one way or another.

What I meant is that it makes more sense to me to enable/disable the entire `cloud_status = repo.status(targets=target, cloud=True)` and not make it specific about `imports`, if that makes sense. 

In your case, there might be something especially slow about imports, but in other cases `cloud_status` would be the part that takes significant time and people might want to make it optional.

So, make it a more straightforward choice for the user: whether to check the remote storage (cloud) or not. Without getting into DVC implementation details like `deps`/`imports`.

>  instead of ~2 seconds, we would need ~10 minutes to generate the DAG with stage statuses.
> I think, actually, checking the status of imported deps should be sped up in the first place.

Could you open an issue sharing some more details on your setup? Those numbers look like too much time

Reply:
> What I meant is that it makes more sense to me to enable/disable the entire `cloud_status = repo.status(targets=target, cloud=True)` and not make it specific about `imports`, if that makes sense. Use that condition to also skip the `import` check as you do know.

Yes, that totally makes sense, and I think it a `-c` flag should be included here. For the reason you mentioned.

But, actually I think there are two things going on here.

As I say in #9304, calling `dvc status` is bad, when having multiple large imports. But, what is even worse in this PR, is where we basically do [this](https://github.com/ostromann/dvc/blob/feature/dag-status/dvc/commands/dag.py#L238):
```
from dvc.repo import Repo
repo = Repo(".")
for stage in repo.index.stages:
    for dep in stage.deps:
       dep.status()
```
Calling the status on the deps of an import stage is horribly slow compared to non-import stages. I'm not sure what could be a smoothest way to get the best of both worlds. Being able to check the the remote storage status, but not having to wait for minutes because you are using imports. 🤷 With having both flags `-c` and `--skip-import-deps` (As I have now renamed it in [52503fd](https://github.com/ostromann/dvc/commit/52503fd0f5e7087783cffe3bbe76390ce83ac037) it would be possible. But I totally understand, it's desirable to keep the option list short and not confuse users.

One alternative I could think of, is to keep the `CmdDAG` class like it was before this PR and introduce a new class `CmdDAGStatus` that can have all the extra fluff 🚀